src/frontend/components/Button.jsx

```diff
@@ -1,19 +1,18 @@
 import React, {Component} from 'react';
 import ReactDOM from 'react-dom';
 
-export default class Button extends Component {
-    constructor(props) {
-        super(props);
-    }
+const Button = props => (
+  <button disabled={props.disabled} type={props.type} id={props.id} onClick={props.onClick}>
+    {props.textContent}
+  </button>
+);
 
-    render() {
-        if(this.props.disabled) {
-            return <button disabled type={this.props.type} id={this.props.id} onClick={this.props.onClick}>
-                {this.props.textContent}
-            </button>
-        }
-        return <button type={this.props.type} id={this.props.id} onClick={this.props.onClick}>
-            {this.props.textContent}
-        </button>
-    }
-}
+Button.propTypes = {
+  disabled: React.PropTypes.bool,
+  textContent: React.PropTypes.string,
+  type: React.PropTypes.string,
+  id: React.PropTypes.string,
+  onClick: React.PropTypes.func
+};
+
+export default Button;
```

Comment:
A few things here:

* Functional component again. 
* The only difference I could see in the 2 branches was `disabled`, so i just bound that prop to the conditional directly. 
* PropTypes also FTW.
* Not sure about the usage of `textContent`? Is that being passed as a regular prop, like `<Button textContent="Foo">`, or is it meant to refer to the child of the component, like `<Button>Foo</Button>`. If it's the latter, then the correct usage would be `{props.children}`, rather than `{props.textContent}`.
* Finally, I'm not really sure what purpose this component serves. What does it do that a regular `button` doesn't do?